plugin/api.go

```diff
@@ -229,6 +229,13 @@ type API interface {
 	// Minimum server version: 5.2
 	UpdateUserStatus(userId, status string) (*model.Status, *model.AppError)
 
+	// UpdateUserStatusWithDNDTimeout will set a user's status until the user, or another integration/plugin, sets it back to online.
+	// The status parameter can be: "online", "away", "dnd with end time", or "offline".
```

Comment:
Is it really "dnd with end time" or just "dnd"? From what I am seeing, the status is still `model.STATUS_DND`. What makes it timed is the `endTime` parameter.

I think we can remove the `status` parameter altogether and just keep this API endpoint for setting timed dnd statuses. Right now, for setting online or away status, there seems to be two APIs to do the same thing. It might confuse users as to which one to use. We can keep using `UpdateUserStatus` for the old stuff, and use the new API purely to set timed DND status. This way we can also get rid of the if condition which checks for DND and uses the new path.